tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts.py

```diff
@@ -780,6 +784,45 @@ def test_add_gs_reference_object_no_md5():
     }
 
 
+def test_add_gs_reference_with_dir_paths():
+    artifact = wandb.Artifact(type="dataset", name="my-folder-arty")
+    mock_gcs(artifact, override_blob_name="my_folder/")
+    artifact.add_reference("gs://my-bucket/my_folder/")
+
+    assert len(artifact.manifest.entries) == 0
+
+
+def test_load_gs_reference_with_dir_paths():
+    artifact = wandb.Artifact(type="dataset", name="my-folder-arty")
+    mock = mock_gcs(artifact, override_blob_name="my_folder/")
+    artifact.add_reference("gs://my-bucket/my_folder/")
```

Comment:
I needed an artifact to pass into this mock when creating it, but I can also just change the `mock_gcs` function to make artifact optional, since its not necessary for this test. open to making that change if anyone has a pref